### ISSUE TITLE
stats.t.fit has no real tests (and several bugs?)

### ISSUE THREAD

**Issue by rohan-shah** [scipy/scipy#8858]
I cannot get stats.t.fit to work, as described on the documentation page at https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.t.html. 

```
data = scipy.stats.t.rvs(df = 10, size = 1000)
print(scipy.stats.t.fit(data))
print(scipy.stats.t.fit(data, scale = 100))
print(scipy.stats.t.fit(data, loc = 100))
print(scipy.stats.t.fit(data, df = 100))```

The error message of the last line is "Unknown arguments: {'df': 100}". Also, the parameters scale and loc of the fit function appear to be ignored, although they're accepted without error.

Suggestions:
Function signature needs to match the help page.
Inputs loc and scale need to be actually used. 
The fit function **needs actual tests**. I couldn't find any in the tests folder, besides generic continuous_rv tests. I would be amazed if the generic tests were sufficient, given that continuous distributions can have wildly different numbers of parameters.

**Comment by pv** [comment]
The parameters provided to fit are starting parameters for fitting, not
fixed paramters.
.
The only issue here appears to be that df cannot be given as a keyword
argument, only positional.

**Comment by josef-pkt** [comment]
I'm surprised that the doc page of the individual distribution don't link to a doc pages for the individual methods. The docs are available for the super class
https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.rv_continuous.fit.html
which is pretty explicit about the arguments to fit.

(generic code mainly needs to be tested generically.)

**Comment by rohan-shah** [comment]
Right, in that case I agree that the only actual bug is that df cannot be specified by keyword. 

I'd like to point out that the distinction between loc and floc is **extremely** counter-intuitive, and not well explained in the documentation. And what the hell is with f0, fn, fa and fb? That part is totally unreadable.

**Comment by josef-pkt** [comment]
You can add more example to the doc of fit, in a PR.

I think the "f-keyword" example is pretty good in explaining it.
But there are no examples for using start_params.

f0, f1, ... predates the ability of using parameter names fa, fb, AFAIR.
